### ISSUE TITLE
Email validator fails on IDNs

### ISSUE THREAD

**Issue by bmorg** [jzaefferer/jquery-validation#1372]
The email validator wrongfully rejects internationalized domain names such as `münchen.de`.

Additionally, the source of the email validator referenced [here](https://github.com/jzaefferer/jquery-validation/blob/master/src/core.js#L1128) is no longer available.

I know email validation is [hard](http://ex-parrot.com/~pdw/Mail-RFC822-Address.html), but I would much rather have a more lenient validator than e.g. preventing a user with a valid email address from signing up.

**Comment by jzaefferer** [comment]
Please see https://github.com/jzaefferer/jquery-validation#reporting-an-issue - which also has a working reference.

**Comment by bmorg** [comment]
Thank you for the quick response!

I suggest also including this information in the [CONTRIBUTING.md](https://github.com/jzaefferer/jquery-validation/blob/master/CONTRIBUTING.md) file, which is linked on the "New Issue" page.

**Comment by bmorg** [comment]
Just for reference, in case somebody is facing similar problems: I created two custom rules to work around the limitations of the built-in email validator.

A rule to allow any set of (non-whitespace) characters as the hostname:
```javascript
jQuery.validator.addMethod("laxEmail", function(value, element) {
    // allow any non-whitespace characters as the host part
    return this.optional( element ) || /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@(?:\S{1,63})$/.test( value );
}, 'Please enter a valid email address.');
```

In my case I needed to only add German umlauts and the "ß" character (`äöüÄÖÜß`):

```javascript
jQuery.validator.addMethod("umlautEmail", function(value, element) {
    // allow German umlauts in the host part
    var let_dig = 'a-zA-Z0-9\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df';
    var re = new RegExp("^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[" + let_dig + "](?:[" + let_dig + "-]{0,61}[" + let_dig + "])?(?:\\.[" + let_dig + "](?:[" + let_dig + "-]{0,61}[" + let_dig + "])?)*$");
    return this.optional(element) || re.test(value);
}, 'Please enter a valid email address.');
```

-----------------------

By the way, the [referenced HTML5 specification](https://html.spec.whatwg.org/multipage/forms.html#valid-e-mail-address) supposedly defines an ABNF, "the character set for which is Unicode", but then continues to define the ABNF for ASCII only.

**Comment by jzaefferer** [comment]
Thank you for the feedback! I've added the warning to the contributing guidelines, along with moving the details for reporting bugs in there as well. I've added references to `addMethod` in both places and used your `laxEmail` as an additional example [there](http://jqueryvalidation.org/jQuery.validator.addMethod/#example:-adds-a-custom-email-validation-method-that-is-less-strict-than-the-one-built-in).
